include/ignition/utils/detail/DefaultOps.hh

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (C) 2018 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#ifndef IGNITION_UTILS_DETAIL_DEFAULTOPS_HH_
+#define IGNITION_UTILS_DETAIL_DEFAULTOPS_HH_
+
+#include <type_traits>
+
+namespace ignition
+{
+  namespace utilities
+  {
+    namespace detail
+    {
+      //////////////////////////////////////////////////
+      template <class T>
+      void VerifyComplete()
+      {
+        // If you are brought here by a compilation error while using the
+        // ImplPtr class or the UniqueImplPtr class, be sure to use the
+        // MakeImpl<T>() or MakeUniqueImpl<T>() function when constructing your
+        // [Unique]ImplPtr instance.
```

Comment:
I think that comment could be helpful as part of the assert message. Not sure how easy it will be for downstream users to find their way back here.